### PR TITLE
test: Add resendwallettransactions functional tests

### PR DESCRIPTION
**Pull request by promag**
Adds functional tests to cover the behaviour introduced in #10995.

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Thanks!

**Comment by TheBlueMatt** [comment]
>You mean restart with explicit arg (`=true`)?

Can do that too, just always a good idea to test failure and success conditions together.

>Maybe only the one calling `resendwallettransactions`?

I just figured if you move all the walletbroadcasting stuff to one place (wallet.py is already big enough) but don't really care.

File: test/functional/resendwallettransactions.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test resendwallettransactions RPC."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
```
Comment (jnewbery) [inline test/functional/resendwallettransactions.py]: nit: prefer explicit import if you're only using one function from test_framework.util

File: test/functional/resendwallettransactions.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    def __init__(self):
+        super().__init__()
+        self.extra_args = [['--walletbroadcast=false']]
+        self.num_nodes = 1
+
+    def run_test(self):
+        # Should raise a RPC error if walletbroadcast is disabled.
+        assert_raises_jsonrpc(-32600, "Error: Wallet transaction broadcasting is disabled with -walletbroadcast", self.nodes[0].resendwallettransactions)
+
+        # Should return an empty array if there aren't mempool transactions.
```
Comment (jnewbery) [inline test/functional/resendwallettransactions.py]: Comment is incorrect. This isn't about transactions in the mempool, it's about transactions in the wallet.

File: test/functional/resendwallettransactions.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        self.stop_node(0)
+        self.nodes[0] = self.start_node(0, self.options.tmpdir)
+        assert_equal(self.nodes[0].resendwallettransactions(), [])
+
+        # Should return an array with the mempool transaction.
```
Comment (jnewbery) [inline test/functional/resendwallettransactions.py]: As above, this isn't about mempool

**Review by jnewbery — COMMENTED** [review]
Each test script incurs about 5 seconds of overhead (node startup and shutdown), so it's much better to group tests for related functionality together into a single script. I think it makes sense to include this test in `wallet.py`.

**Comment by promag** [comment]
Thanks @jnewbery, comments addressed, and rebased with #11002.

Regarding where to place the tests, IMO performance issues shouldn't lead to long-and-hard-to-follow-and-also-tedious-to-change tests. It also makes parallel tests hard to implement. Beside that, in this particular case, at least one restart must occur.

**Comment by jnewbery** [comment]
@promag and I chatted about this. I still prefer combining this with `wallet.py` to avoid additional test setup/teardown overhead (if it can be done in a way that doesn't make the test less clear or maintainable).

I'm happy to defer to consensus opinion if other people disagree.

**Comment by sdaftuar** [comment]
ACK f9853283623df7ab3972611005b9dd089b6985ca

@jnewbery  I don't like long test-suite run times either, but in this case merging a test with the already hard-to-read `wallet.py` seems insufficiently compelling, especially since this is already done and working and is an easy test to read/understand/maintain.

**Comment by jnewbery** [comment]
@sdaftuar - sure, this is very much a case-by-case judgement call. Perhaps Matt's suggestion of having a separate test for all the wallet broadcast functionality is the best way to organise these tests.

In any case, I certainly won't NACK any new tests based on my personal taste, and we can always refactor in later PRs if it makes sense to group things differently.

**Comment by promag** [comment]
Rebased since #11002 is merged.

**Pull request merged**